### PR TITLE
[fix #153] cdc: fix buffer sink

### LINKED ISSUE #153: cdc: data lost
### 2. Minimal reproduce step (Required)
1. start src tikv and dst tikv
2. deploy the cdc synchronization task
3. put key into src tikv
4. read key from dst tikv

### 3. What did you see instead (Required)
Can't read key from dst tikv

### 4. What did you expect to see? (Required)
read key from dst tikv

### 5. What is your migration tool and TiKV version? (Required)

- TiKV CDC:

### PR DESCRIPTION
**Pull request by zeminzhou**
Signed-off-by: zeminzhou <zhouzemin@pingcap.com>

### What problem does this PR solve?
Some entry which cmmit_ts greater then resolved ts may send to dst tikv

Issue Number: close #153

Problem Description: **TBD**
Some entry which cmmit_ts greater than resolved ts may send to dst tikv

### What is changed and how does it work?
Only send entry which commit_ts less than resolved ts

### Code changes

- Has exported function/method change
- Has exported variable/fields change
- Has interface methods change
- Has persistent data change
- No code

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Unit test

### Side effects
- No side effects

### CODE REVIEW

**Review by haojinming — APPROVED** [review]
LGTM~

**Review by pingyu — COMMENTED** [review]
Please add related unit test. Rest LGTM.

**Comment by zeminzhou** [comment]
> Please add related unit test. Rest LGTM.



> Please add related unit test. Rest LGTM.

Add unit test in New PR.

**Comment by zeminzhou** [comment]
> Please add related unit test. Rest LGTM.

Done

File: cdc/cdc/sink/buffer_sink_test.go [hunk 1]
```diff
@@ -33,6 +33,100 @@ func TestKeySpanIsNotFlushed(t *testing.T) {
...
+	b.bufferMu.Lock()
+	require.Equal(t, 5, len(b.buffer[uint64(2)]))
+	b.bufferMu.Unlock()
+
+	_, err := b.FlushChangedEvents(ctx, 1, 4)
+	require.Nil(t, err)
+	retry := 0
+	for retry < 5 {
+		b.bufferMu.Lock()
+		if backendSink.accumulated == 3 && len(b.buffer[uint64(1)]) == 2 {
```
Comment (pingyu) [inline cdc/cdc/sink/buffer_sink_test.go:74]: Suggest to copy value of `backendSink.accumulated` & `len(b.buffer[uint64(1)])` inside lock & judge outside.

**Review by pingyu — COMMENTED** [review]
Rest LGTM~

File: cdc/cdc/sink/buffer_sink_test.go (hunk 1 above)
Comment (zeminzhou) [reply cdc/cdc/sink/buffer_sink_test.go]: Done

Comment (zeminzhou) [reply cdc/cdc/sink/buffer_sink_test.go]: Done

**Review by pingyu — APPROVED** [review]
LGTM~

**Pull request merged**